src/AmalgamatorSession.ts

```diff
@@ -488,6 +498,56 @@ export class AmalgamatorSession extends LoggingDebugSession {
         }
     }
 
+    protected async disassembleRequest(
+        response: DebugProtocol.DisassembleResponse,
+        args: DebugProtocol.DisassembleArguments
+    ): Promise<void> {
+        if (args.memoryReference) {
+            response.body = {
+                instructions: [],
+            };
+            try {
+                this.addressMap.forEach(
+                    (
+                        [instructionPointerReference, startAdress, endAddress],
+                        childDapID
+                    ) => {
+                        if (
+                            args.memoryReference ===
+                                instructionPointerReference ||
+                            args.memoryReference === startAdress ||
+                            args.memoryReference === endAddress
+                        ) {
+                            this.childDapIndex = childDapID;
+                        }
+                    }
+                );
+                const disassemble = await this.childDaps[
+                    this.childDapIndex as number
+                ].disassembleRequest(args);
+                response.body = disassemble.body;
+                this.addressMap.set(this.childDapIndex as number, [
+                    this.addressMap.get(
+                        this.childDapIndex as number
+                    )?.[0] as string,
+                    disassemble.body?.instructions[0].address as string,
+                    disassemble.body?.instructions[
+                        disassemble.body?.instructions.length - 1
+                    ].address as string,
+                ]);
```

Comment:
Doing my suggestion at line 411 makes this simpler too:

```typescript
const instructions = disassemble.body?.instructions;
this.addressMap.set(instructions[0].address, this.childDapIndex);
this.addressMap.set(instructions[instructions.length - 1].address, this.childDapIndex);
```
Although I don't think you should use `body?` here and instead error if body is empty.

Reply:
I have updated it.